lib/extensions/ar_virtual.rb

```diff
@@ -39,6 +39,9 @@ def arel_attribute(column_name, arel_table = self.arel_table)
       end
     end
 
+    def attribute_in_sql?(column_name)
+      !virtual_attribute?(column_name) || _virtual_arel[column_name.to_s]
```

Comment:
I believe this will return `true` if you give it anything that's not a virtual attribute, i.e. a method that's not virtual, typo, etc..